cql3/restrictions/statement_restrictions.cc

```diff
@@ -363,14 +363,14 @@ statement_restrictions::statement_restrictions(database& db,
                                     "(unless you use the token() function or allow filtering)");
                         }
                         _partition_key_restrictions = _partition_key_restrictions->merge_to(_schema, restriction);
+                        _partition_range_is_simple &= !find(restriction->expression, expr::oper_t::IN);
```

Comment:
Is there any impact (performance, correctness) to the bug? What happens if we get it wrong (say, have a clustering restriction with IN? 

We may need to backport.

Reply:
There's definitely impact; the bug was discovered when this code was used for a slightly different purpose that has test coverage for clustering IN.  I'll file a bug if I can demonstrate incorrect behaviour prior to this change.

Reply:
The only previous impact is on performance.  The bug was introduced in 4661aa026; it doesn't look like a backport is needed.

Reply:
Please add this information to the changelog.

Reply:
Done.